src/input.ts

```diff
@@ -177,16 +268,61 @@ export class Mouse {
     this._keyboard = this._page.keyboard;
   }
 
+  async _updateMouseIndicator() {
+    if (!this._page._browserContext._browserBase._options.showMouseIndicator)
+      return;
+    const args = [this._x, this._y, !!this._buttons.size] as const;
+    (await this._page.mainFrame()._utilityContext()).evaluateInternal(([x, y, pressed]) => {
+      const element: HTMLElement = document.querySelector('body > playwright-mouse-indicator') || createMouseIndicator();
+      element.style.left = x + 'px';
+      element.style.top = y + 'px';
+      element.classList.toggle('pressed', pressed);
+      window.getComputedStyle(element).fill;
+
+
+      function createMouseIndicator() {
+        const element = document.createElement('playwright-mouse-indicator');
+        const shadow = element.attachShadow({mode: 'closed'});
+        shadow.innerHTML = `
+        <style>
+          svg {
+            filter: drop-shadow(0px 2px 2px rgba(0,0,0,0.3));
+            opacity: 0.5;
+          }
+          polygon {
+            fill: white;
+            stroke: black;
+            transition: fill .2s;
+          }
+          :host(.pressed) polygon {
+            transition: none;
+            fill: black;
+            transition: none;
+          }
+        </style>
+        <svg width="40" height="40">
+          <polygon points="0,0 0,34 8,28 14,38 20,35 14,26 24,24"/>
+        </svg>`;
+        element.style.position = 'fixed';
+        element.style.pointerEvents = 'none';
+        element.style.zIndex = String(Number.MAX_SAFE_INTEGER - 1);
+        document.body.appendChild(element);
```

Comment:
nit: how about attaching to `documentElement` instead of `body`? Feels like it'll be less invasive.